### DISCUSSION TITLE
Can not get wallet from StoredKey

### DISCUSSION THREAD

**Discussion by tuannm2** [trustwallet/wallet-core#3286] [Q&A]
**Describe the bug**
When I import private key using function:
```
StoredKey.importPrivateKey(privateKey.data(), name, password.toByteArray(), coin);
```

And then export private key like this:
```
val wallet = storedKey.wallet(password.toByteArray());

val privateKey = wallet.getKey(account.coin(), account.derivationPath());
```
The password return wrong

However when I export private key by this function:
```
val privateKey = storedKey.privateKey(account.coin(), password.toByteArray());
```
It work fine.

The password are the same.

Can you explain this. Thank you

**Comment by weixuefeng** [comment]
`storedKey.wallet` will return `HDWallet`, but HDWallet need `mnemonic`, you only import privateKey, not mnemonic.
if you need wallet, you should call `StoredKey.importHDWallet()`.
